### PR TITLE
Example app

### PR DESCRIPTION
**Pull request by btkelly**
Added to the example application so demonstrate Gandalf while using the GandalfActivity. This also uses MockWebServer to allow the user to restart the application in each of the different states. This does not include Dialogs as that is part of another card, working on that next. Fixes #27

### CODE REVIEW

File: example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java [hunk 1]
```diff
@@ -15,8 +15,85 @@
...
+import io.github.btkelly.gandalf.activities.GandalfActivity;
+import io.github.btkelly.gandalf.example.R;
+import io.github.btkelly.gandalf.example.utils.MockWebServerUtil;
+
 /**
- * TODO: Add a class header comment!
+ * An example splash activity that demonstrates the simplest way to add Gandalf to a project
  */
-public class SplashActivity {
+public class SplashActivity extends GandalfActivity {
```
Comment (jsibbold) [inline example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java]: I recommend we name this GandalfSplashActivity, just to distinguish it as we (and likely almost everyone else) always name our own SplashActivity the exact same thing. I know it's long but I think in this case it makes sense

Comment (jsibbold) [inline example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java]: nvm probably fine this way

Comment (btkelly) [inline example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java]: Haha I like this, I'll just wait on all PR comments until people change their mind

**Comment by jsibbold** [comment]
Approved

**Pull request merged**